docs/MessageTokenizer.md

```diff
@@ -0,0 +1,260 @@
+```
+
+Otherwise, return the character at `currentIndex` and step forward one (1).
+
+### hasNextSequence(String)
+
+> Return Type: boolean
+>
+> Throws: Nothing
+
+Returns `true` if the remaining content has the specified String; otherwise `false`.
+
+### hasNextWord()
+
+> Return Type: boolean
+> 
+> Throws: Nothing
+
+Same as `hasNext()`
+
+### nextWord()
+> Return Type: MessageTokenizer.Token
+>
+> Throws: IllegalStateException
+
+```java
+    if(!this.hasNextWord()) {
+      throw new IllegalStateException("No more words found!");
+    }
+```
+Otherwise, grabs the next word in the sequence.
+
+### hasNextLine()
+
+> Return Type: boolean
+> 
+> Throws: Nothing
+
+Same as `hasNext()`
+
+
+### nextLine()
+
+> Return Type: MessageTokenizer.Token
+>
+> Throws: IllegalStateException
+
+```java
+  if(!this.hasNextLine()) {
+      throw new IllegalStateException("No more lines found!");
+    }
+```
+
+Otherwise, grabs the next line in the sequence.
+
+### hasNextPattern(Pattern)
+
+> Return Type: boolean
+> 
+> Throws: Nothing
+
+`true` if `hasNext()` is true **and** the given pattern has a match in the remaining content.
+
+### nextPattern(Pattern)
+
+> Return Type: MessageTokenizer.Token
+>
+> Throws: IllegalStateException
+
+```java
+    if(!this.hasNextRegex(pattern)) {
+      throw new IllegalStateException("No more occurrences found!");
+    }
+```
+```java
+      if(!matcher.find()) {
+        throw new IllegalStateException("Couldn't find any matches!");
+      }
+```
+ 
+ Otherwise, returns the regex in the content and steps to the end of it.
+
+### hasNextInvite()
+
+> Return Type: boolean
+>
+> Throws: Nothing
+
+`true` if there's an invite in the message, otherwise `false`.
+
+### nextInvite()
+
+> Return Type: MessageTokenizer.InviteToken
+> 
+> Throws: IllegalStateException
+
+```java
+    if(!this.hasNextInvite()) {
+      throw new IllegalStateException("No more invites found!");
+    }
+```
+
+```java
+      if(!matcher.find()) {
+        throw new IllegalStateException("Couldn't find any matches!");
+      }
+```
+
+Otherwise, returns the next Invite found (Example: `discord.gg/NxGAeCY`) and steps to the end of it.
+
+### hasNextMention()
+
+> Return Type: boolean
+>
+> Throws: Nothing
+
+`true` if there is another mention (channel, user, role, etc..) in the message, otherwise `false`
+
+### nextMention()
+
+> Return Type: MessageTokenizer.MentionToken
+>
+> Throws: IllegalStateException
+
+```java
+    if(!this.hasNextMention()) {
+      throw new IllegalStateException("No more mentions found!");
+    }
+```
+
+```java
+      } else {
+        throw new IllegalStateException("Couldn't find a mention even though it was found!");
+      }
+```
+
+Otherwise, returns a `MessageTokenizer.RoleMentionToken`, `MessageTokenizer.UserMentionToken`, or a `MessageTokenizer.ChannelMentionToken`.
+
+### hasNextEmoji()
+
+> Return Type: boolean
+>
+> Throws: Nothing
+
+`true` if there is an emoji in the remaining content, otherwise `false`
```

Comment:
You should specifically state it returns only custom emoji and not Unicode emoji,